src/types.rs

```diff
@@ -1620,20 +2075,102 @@ pub struct RuleContext {
 /// A first-order term rewriting system.
 #[derive(Debug, Hash, Clone, PartialEq, Eq)]
 pub struct TRS {
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Rule, parse_rule, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    /// 
+    /// // Constructing a TRS manually 
+    /// let r0 = parse_rule(&mut sig, "A(x_) = A(B)").expect("parse of A(x_) = A(B)");
+    /// let r1 = parse_rule(&mut sig, "B = C | D").expect("parse of B = C | D");
+    /// let r2 = parse_rule(&mut sig, "E(F) = G").expect("parse of E(F) = G");
+    /// 
+    /// let t = TRS::new(vec![r0, r1, r2]);
+    ///
+    /// // Constructing a TRS using the parser
+    /// let t = parse_trs(&mut sig, "A(x_) = A(B);\nB = C | D;\nE(F) = G;")
+    ///     .expect("parse of A(x_) = A(B); B = C | D; E(F) = G;");
+    /// 
+    /// // or alternatively for readability of the TRS
+    /// let t = parse_trs(&mut sig, 
+    /// "A(x_) = A(B);
+    /// B = C | D;
+    /// E(F) = G;").expect("parse of A(x_) = A(B); B = C | D; E(F) = G;");
+    /// ```
```

Comment:
As with `Rule`, this example should go outside the struct definition, and some short comments should describe what each field does. Could you please change to reflect that?